### DISCUSSION TITLE
Support documentation hover for function parameters

### DISCUSSION THREAD

**Discussion by pokey** [microsoft/pylance-release#2653] [Enhancement]
In typescript, I can hover over a parameter, and see the documentation from the js doc.  However, in Python I just get type information:

<img width="367" alt="parameter-hints" src="https://user-images.githubusercontent.com/755842/144869234-ad677aac-6bd2-45d4-a3a8-f87af8af5cf2.png">

I would expect the hover to have something like

```
(parameter) bar: str

The bar is in town
```

I'm not sure if this is supposed to be supported, or if there's just something broken in my setup?  Apologies if there's already an issue for this one; I would think I'm not the first person to want this 😅.  Did a bunch of searching and didn't find a good match, though

VSCode version info:
```
Version: 1.62.3
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 19.6.0
```

Pylance version info:
```
Name: Pylance
Id: ms-python.vscode-pylance
Description: A performant, feature-rich language server for Python in VS Code
Version: 2021.12.0
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.vscode-pylance
```

In case helpful, code in the above screenshot:

```py
def foo(bar: str) -> str:
    """Something

    Args:
        bar (str): The bar is in town

    Returns:
        str: Info about the bar
    """
    return bar + "hello"
```

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by jtrinklein** [comment]
I would definitely like to see this support as well. Supporting some sort of standard markup like [sphynx markup](https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#python-signatures) (which is apparently the default style used for pycharm's comment autogeneration) would be a massive improvement.

SO answer for reference: https://stackoverflow.com/a/40596167

**Comment by rattrayalex** [comment]
@judej would you be open to PRs for this? If so, would supporting a handful of common argument formats work for you?
